### ISSUE TITLE
Part 7: Decomposition, part one, why dont answer initial question?

### ISSUE THREAD

**Issue by XinyueZ** [langchain-ai/rag-from-scratch#6]
Hey,

```python
question = "What are the main components of an LLM-powered autonomous agent system?"
questions = generate_queries_decomposition.invoke({"question":question})
```

My question is, why don't we put the original question into the QA chain later?

`questions.append(question) or [question].extend(questions)`

BR

**Comment by pankajrawat9075** [comment]
Yes that's correct question and we have to append the original question to the last.
Just use : questions = [q for q in questions[1:] if q] + [question]

Linked image from paper (least to most) where it's mentioned.
Thanks
![image](https://github.com/user-attachments/assets/9949c526-a200-4041-b347-8bb58b617cca)
